[Soh Tze Jen] iP

Meatsushi64 opened the pull request:
### Jen Chatbot

> Simple is good, less is more

Jen helps you manage tasks. 🤲 It is:
- simple
- fast
- free
Did i mention that it is:

1. _FREE???_

Download it **now** and let it manage your tasks for you!
Features
- [ ] Managing Tasks
- [ ] Managing Deadlines
- [ ] Managing Events
- [ ] Save to file

My mental state when programming this: [https://youtu.be/VKMw2it8dQY?si=6MtuxqVS-1H4QiMo](url)

If you use Java, the chatbot starts with the Jen.java class. Here is the main method:
`    public static void main(String[] args) {
        Jen chatbot = new Jen();
        chatbot.start();
    }`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Meatsushi64 committed:
Renamed bot to Jen and changed greeting

Meatsushi64 committed:
Removed extra space before greeting message

Meatsushi64 committed:
Added Msg, Parser, UI classes

Meatsushi64 committed:
Added Command classes and Command handling to the rest of the classes. Storage class added to handle tasks

Meatsushi64 committed:
Added Unmark and Mark commands, with handling by other classes

Meatsushi64 committed:
Added task display after marking and umarking task

Meatsushi64 committed:
Added task types and handlers in Parser class

Meatsushi64 committed:
modified message and updated testing file

Meatsushi64 committed:
Updated to handle errors in parsing

Meatsushi64 committed:
Added delete feature to task, and handle exceptions with delete, mark and unmark commands

Meatsushi64 committed:
added enum for command types

Meatsushi64 committed:
Edit Parser to follow coding standard

Meatsushi64 committed:
Add ability to write to save file

Meatsushi64 committed:
Add date format to Deadline

Meatsushi64 committed:
Merge branch 'branch-Level-8'

Meatsushi64 committed:
Organise to packages

Meatsushi64 committed:
Merge branch 'branch-A-Packages'

Meatsushi64 committed:
Merge branch 'add-gradle-support'

Meatsushi64 committed:
Add gradle support and configs

Meatsushi64 committed:
Add test cases for Save and Storage

Meatsushi64 committed:
Merge branch 'A-JUnit'

Meatsushi64 committed:
Configure gradle build to create jar file

Meatsushi64 committed:
Merge branch 'branch-A-Jar'

Meatsushi64 committed:
Add JavaDocs to all files

Meatsushi64 committed:
Merge branch 'branch-A-JavaDoc'

Meatsushi64 committed:
Edit Jen.java to coding style

Meatsushi64 committed:
Fix merge issue

Meatsushi64 committed:
Update to Java Coding Standard

Meatsushi64 committed:
Merge branch 'branch-A-CodingStandard'

Meatsushi64 committed:
Add Find Command

Meatsushi64 committed:
Merge branch 'branch-Level-9'

Meatsushi64 committed:
Edit JavaDocs

Meatsushi64 committed:
Add CheckStyle and make necessary edits

Meatsushi64 committed:
Add GUI

Meatsushi64 committed:
Merge branch 'branch-Level-10'

ceilingFan456 commented on src/main/java/jen/tasks/ToDo.java:
```diff
@@ -0,0 +1,25 @@
+     *
+     * @param description The description of the to-do task.
+     */
+    public ToDo(String description) {
```
I think it would look nicer if there is a space here.

ceilingFan456 commented on src/main/java/jen/Storage.java:
```diff
@@ -0,0 +1,133 @@
+
+    /**
+     * Adds the current item to the ArrayList of Tasks.
+     * It increments the size, which represents the number of items in the list
```
it would be more consistent if there is a period here.

ceilingFan456 commented on src/main/java/jen/Storage.java:
```diff
@@ -0,0 +1,133 @@
+    }
+
+    /**
+     * Deletes the task at index (i - 1) of the list. ArrayList starts at index 0, so when user
```
It would be more consistent if periods are added here.

ceilingFan456 commented on src/main/java/jen/Storage.java:
```diff
@@ -0,0 +1,133 @@
+    }
+
+    /**
+     * Checks whether the index i is within the number
```
The code might look cleaner if there is a space between description and parameters. might consider adding the space for the rest of the javadoc in this class too.

ceilingFan456 reviewed (commented):
I really like your code! it is clear and simple to read. Only a few minute style issues but they do not pose any problem understanding the code at all.

lyhthaddeus commented on src/main/java/jen/JenException.java:
```diff
@@ -0,0 +1,17 @@
+ * Represents a custom exception specific to the Jen chatbot.
+ * This exception is thrown when the chatbot encounters an error during execution.
+ */
+public class JenException extends Exception {
```
is this exception really neccessary?

lyhthaddeus commented on src/main/java/jen/MainWindow.java:
```diff
@@ -0,0 +1,57 @@
+/**
+ * Controller for the main GUI.
+ */
+public class MainWindow extends AnchorPane {
```
Could all these Ui javafx classes be in its own package?

lyhthaddeus commented on src/main/java/jen/Msg.java:
```diff
@@ -0,0 +1,14 @@
+ * Provides predefined messages used by the chatbot.
+ * This class contains constants for commonly used chatbot messages.
+ */
+public class Msg {
```
is a class really better than an enum here, since you have no methods?

lyhthaddeus commented on src/main/java/jen/OutOfIndexException.java:
```diff
@@ -0,0 +1,16 @@
+ * Represents an exception that is thrown when an index is out of bounds.
+ * This exception is used to handle cases where an invalid index is accessed in the Jen chatbot.
+ */
+public class OutOfIndexException extends Exception {
```
Could all the exceptions be in one package?

lyhthaddeus commented on src/main/java/jen/Parser.java:
```diff
@@ -0,0 +1,178 @@
+     * @return The corresponding {@code Command} object.
+     * @throws JenException If the input is invalid or unrecognized.
+     */
+    public Command read(String input) throws JenException {
```
I love the clear case by case handling :). regex matching could also be an alternative

lyhthaddeus commented on src/main/java/jen/UI.java:
```diff
@@ -0,0 +1,80 @@
+    /**
+     * Prints the chatbot's greeting message.
+     */
+    public void greet() { // prints greeting
```
Is this comment really needed?

lyhthaddeus commented on src/main/java/jen/UI.java:
```diff
@@ -0,0 +1,80 @@
+     * Prints the chatbot's greeting message.
+     */
+    public void greet() { // prints greeting
+        System.out.println(Msg.LINE);
```
could the code be truncated?
```Suggestion
System.out.println(Msg.LINE + Msg.GREETING + Msg.LINE);
```

lyhthaddeus reviewed (commented):
Overall nice code. I think you could do with more packages for easier navigation :)

Meatsushi64 committed:
Add assertions to Parse.java

Meatsushi64 committed:
Merge branch 'branch-A-Assertions'

Meatsushi64 committed:
Implement SLAP to parse ToDo Command

Low level abstraction mixed into read() method
in Parser.java. read() method beyond 30LoC.

parseTodoInput method added to Parser.java

Abstract out the method to another method in
Parser.java, so that read() is shorter and
functions as a handler for command word

Meatsushi64 committed:
Resolve merge conflicts

Meatsushi64 committed:
Make Parser SLAP

Meatsushi64 committed:
Make Jen SLAP

Meatsushi64 committed:
Merge branch 'branch-A-CodeQuality'

Meatsushi64 committed:
Fix Unmark Command

Meatsushi64 committed:
Add note to tasks

Meatsushi64 committed:
Merge branch 'branch-D-Notes'

Meatsushi64 committed:
Add greeting to Ui

Meatsushi64 committed:
Update README.md

Meatsushi64 committed:
Fix deadline parsing of save file

Meatsushi64 committed:
Merge branch 'master' of https://github.com/Meatsushi64/ip

Meatsushi64 committed:
Fix Javadocs
